rtt-target/src/defmt.rs

```diff
@@ -32,14 +32,18 @@ unsafe impl defmt::Logger for Logger {
         unsafe { CS_RESTORE = restore };
 
         // safety: accessing the `static mut` is OK because we have disabled interrupts.
-        unsafe { ENCODER.start_frame(do_write) }
+        unsafe {
+            let encoder = &mut *core::ptr::addr_of_mut!(ENCODER);
```

Comment:
Why do you do the pointer shenanignas? (consider me inept)

Reply:
Rustc warns on static mut access